src/app/pages/storage/volumes/datasets/dataset-acl/dataset-acl.component.ts

```diff
@@ -24,6 +24,21 @@ import { EntityJobComponent } from '../../../../common/entity/entity-job/entity-
 import {EntityUtils} from '../../../../common/entity/utils';
 import { DialogFormConfiguration } from 'app/pages/common/entity/entity-dialog/dialog-form-configuration.interface';
 
+interface Acl {
+  perms: object;
+  flags: object;
+  tag: string;
+  type: string;
+  id: number;
+}
```

Comment:
`perms` and `flags` have values as attribute names. Which can be generic and change. That's why I kept it as a simple `object`

Reply:
If you mean that `perms` and `flags` look like this:
```ts
{
  name1: string;
  name2: string;
}
```
you can change type to:
```ts
perms: { [name: string]: string };
flags: { [name: string]: string };
```

This will make it more type safe.

Reply:
Looks better!
Let's also make them enums, so that we can address each flag and permission individually:
```typescript
export enum AclFlag {
  FileInherit = 'FILE_INHERIT',
  ...
}

export type AclFlags = { [K in AclFlag]: boolean };

```